pkg/master/master_test.go

```diff
@@ -531,54 +527,62 @@ func testInstallThirdPartyAPIListVersion(t *testing.T, version string) {
 		},
 	}
 	for _, test := range tests {
-		_, fakeClient, server, assert := initThirdParty(t, version)
-		defer server.Close()
-
-		if test.items == nil {
-			fakeClient.ExpectNotFoundGet(etcdtest.PathPrefix() + "/ThirdPartyResourceData/company.com/foos/default")
-		} else {
-			setupEtcdList(fakeClient, "/ThirdPartyResourceData/company.com/foos/default", test.items)
-		}
+		func() {
+			_, etcdserver, server, assert := initThirdParty(t, version)
+			defer server.Close()
+			defer etcdserver.Terminate(t)
+
+			client := etcdserver.Client
+			if test.items != nil {
+				setupEtcdList(client, "/ThirdPartyResourceData/company.com/foos/default", test.items)
+			}
 
-		resp, err := http.Get(server.URL + "/apis/company.com/" + version + "/namespaces/default/foos")
-		if !assert.NoError(err) {
-			return
-		}
-		defer resp.Body.Close()
+			resp, err := http.Get(server.URL + "/apis/company.com/" + version + "/namespaces/default/foos")
+			if !assert.NoError(err) {
+				return
+			}
+			defer resp.Body.Close()
 
-		assert.Equal(http.StatusOK, resp.StatusCode)
+			assert.Equal(http.StatusOK, resp.StatusCode)
 
-		data, err := ioutil.ReadAll(resp.Body)
-		assert.NoError(err)
+			data, err := ioutil.ReadAll(resp.Body)
+			assert.NoError(err)
 
-		list := FooList{}
-		if err = json.Unmarshal(data, &list); err != nil {
-			t.Errorf("unexpected error: %v", err)
-		}
+			list := FooList{}
+			if err = json.Unmarshal(data, &list); err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
 
-		if test.items == nil {
-			if len(list.Items) != 0 {
-				t.Errorf("expected no items, saw: %v", list.Items)
+			if test.items == nil {
+				if len(list.Items) != 0 {
+					t.Errorf("expected no items, saw: %v", list.Items)
+				}
+				return
 			}
-			continue
-		}
 
-		if len(list.Items) != len(test.items) {
-			t.Errorf("unexpected length: %d vs %d", len(list.Items), len(test.items))
-			continue
-		}
-		for ix := range list.Items {
-			// Copy things that are set dynamically on the server
-			expectedObj := test.items[ix]
-			expectedObj.SelfLink = list.Items[ix].SelfLink
-			expectedObj.Namespace = list.Items[ix].Namespace
-			expectedObj.UID = list.Items[ix].UID
-			expectedObj.CreationTimestamp = list.Items[ix].CreationTimestamp
-
-			if !reflect.DeepEqual(list.Items[ix], expectedObj) {
-				t.Errorf("expected:\n%#v\nsaw:\n%#v\n", expectedObj, list.Items[ix])
+			if len(list.Items) != len(test.items) {
+				t.Errorf("unexpected length: %d vs %d", len(list.Items), len(test.items))
+				return
 			}
-		}
+			// The order of elements in LIST is not guaranteed.
+			mapping := make(map[string]int)
+			for ix := range test.items {
```

Comment:
Not really - before when we were using FakeEtcdClient - we were guaranteed the order of items in the returned list. This is no longer the case with real etcd - so we need to sort them before comparing them - there is a comment added above.